### ISSUE TITLE
Feature request: add ListEmptyComponent to List

### ISSUE THREAD

**Issue by kbrandwijk** [react-native-training/react-native-elements#766]
I use the `List` component a lot, because it's easier to specify ListItems than specify a datasource on RN List implementations. However, I am really missing `ListEmptyComponent` on List.

**Comment by iRoachie** [comment]
Hey @kbrandwijk the `List` component for react-native-elements is just an wrapper `View` around your content, so it doesn't actually have any props that affect the List itself. Instead you can use the `FlatList` component from react-native and place it inside our `List`.  You can then use the `ListEmptyComponent` prop on FlatList.

Here's an example snack you can see https://snack.expo.io/B1J4Qvwbf

**Comment by kbrandwijk** [comment]
I appreciate the example, but it's missing an important point I made:
>  it's easier to specify ListItems than specify a datasource on RN List implementations

So let me rephrase: even though the `List` component is just a wrapper View, it would be nice if there was an easy way to specify something that behaves similar to `ListEmptyComponent` without having to use a RN List type.

**Comment by iRoachie** [comment]
You're correct! I was just about to type that :D The `FlatList` can implement a `ListEmptyComponent` is cause it takes in a `data` prop, so it knows when the length of the items in the list its 0. 

Our List is only a wrapper and doesn't affect the rendering/displaying of rows in any way. The only prop it inherently knows about is `children`. I guess we could try something like checking the length of `this.props.children`

**Comment by kbrandwijk** [comment]
I think that wouldn't be a solid solution. The only proper option in my opinion would be to add a listEmpty boolean property along with ListEmptyComponent, but I don't know if there's enough of a gain to be gotten from that. Otherwise, I would just close this.

**Comment by iRoachie** [comment]
I guess that could work, but it'd be the same as doing a conditional check anyway, so not sure it's there's any benefit from it

**Comment by kbrandwijk** [comment]
That's exactly what I meant. If you're okay with it, I think this should be closed.
